### PR TITLE
Renaming variable names in Simulation to avoid ambiguity

### LINKED ISSUE #23037: THMProblem multiple inheritance ambiguous member variables
## Bug Description

THMProblem has a multiple inheritance from both `FEProblem` and `Simulation`. Both base classes define member variables with the same name, including `_mesh`, `_factory`, and `_pars`.
This causes member variable ambiguity in THMProblem and moose apps derived from THMProblem. 

## Steps to Reproduce

In THMProblem or a THMProblem-derived problem, accessing member variables such as `_mesh`, `_factory`, and `_pars` will trigger a compiling error complaining member variable ambiguity.

PS: currently in THMProblem, these variables are not accessed by any function, so it has never been triggered.

## Impact

The bug/inconvenience could be avoided by calling, for example, `FEProblem::_pars' instead of `_pars`, but from a code design perspective, this shouldn't be the case.

To mitigate the issue, it is recommended to use different names for these member variables, for example:

```
{
  THMMesh & _thm_mesh;
  ThermalHydraulicsApp & _thm_app;
  Factory & _thm_factory;
  const InputParameters & _thm_pars;
}
```

### PR DESCRIPTION
**Pull request by lingzou**
## Reason

See issue #23037, this pull request fixes the member variable ambiguity in THMProblem.

## Design

Use different variable names in `Simulation` to avoid variable ambiguity in `THMProblem`, which inherits from both `FEProblem` and `Simulation`.

## Impact

Avoid variables ambiguity in `THMProblem` and classes inherited from `THMProblem`.

Closes #23037

### CODE REVIEW

File: modules/thermal_hydraulics/include/base/Simulation.h [hunk 1]
```diff
@@ -335,16 +335,16 @@ class Simulation : public libMesh::ParallelObject, public LoggingInterface, publ
...
   /// Pointer to FEProblem representing this simulation
   FEProblemBase & _fe_problem;
 
   /// The application this is associated with
-  ThermalHydraulicsApp & _app;
+  ThermalHydraulicsApp & _thm_app;
 
   /// The Factory associated with the MooseApp
-  Factory & _factory;
+  Factory & _m_factory;
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/include/base/Simulation.h:347]: I don't know what the "m" means here - just rename to `_thm_factory`.

**Review by joshuahansel — CHANGES_REQUESTED** [review]
Looks good, just requesting the one rename.

File: modules/thermal_hydraulics/include/base/Simulation.h (hunk 1 above)
Comment (lingzou) [reply modules/thermal_hydraulics/include/base/Simulation.h]: @joshuahansel thanks, the PR has been updated.

Comment (lingzou) [reply modules/thermal_hydraulics/include/base/Simulation.h]: @joshuahansel The PR should be ready if you don't have additional comments/suggestions.

Comment (joshuahansel) [reply modules/thermal_hydraulics/include/base/Simulation.h]: Yes, I'm just invalidating the failing python target and then I'll merge if that passes.

**Review by joshuahansel — APPROVED** [review]

**Pull request merged**